src/components/trace/hat-script-graph.ts

```diff
@@ -69,72 +77,30 @@ class HatScriptGraph extends LitElement {
 
   private render_trigger(config: Trigger, i: number) {
     const path = `trigger/${i}`;
-    const tracked = this.trace && path in this.trace.trace;
+    const track = this.trace && path in this.trace.trace;
     this.renderedNodes[path] = { config, path };
-    if (tracked) {
+    if (track) {
       this.trackedNodes[path] = this.renderedNodes[path];
     }
     return html`
       <hat-graph-node
         graphStart
+        ?track=${track}
         @focus=${this.selectNode(config, path)}
-        class=${classMap({
-          track: tracked,
-          active: this.selected === path,
-        })}
+        ?active=${this.selected === path}
         .iconPath=${mdiAsterisk}
-        tabindex=${tracked ? "0" : "-1"}
+        tabindex=${track ? "0" : "-1"}
       ></hat-graph-node>
     `;
   }
 
   private render_condition(config: Condition, i: number) {
     const path = `condition/${i}`;
-    const trace = this.trace.trace[path] as ConditionTraceStep[] | undefined;
-    const track_path =
-      trace?.[0].result === undefined ? 0 : trace[0].result.result ? 1 : 2;
     this.renderedNodes[path] = { config, path };
-    if (trace) {
+    if (this.trace && path in this.trace.trace) {
       this.trackedNodes[path] = this.renderedNodes[path];
```

Comment:
This is pretty much the only place (except the choose node) where trackedNodes is modified in a renderer...

Come to think of it, the choose node could probably be made to better recursively use `render_node` which would make render_condition the only special case for this. It may be worth trying to refactor this out as well and just use `render_condition_node` directly instead.

Reply:
`render_node` does the same right? Also adding nodes during rendering? I don't really see the benefit here?

Reply:
I was just thinking it's an almost unique side effect to modify trackedNodes here, which I think is cosidered a code smell?

Reply:
Yeah... I think it is ok here...